### ISSUE TITLE
Clawpack code issues

### ISSUE THREAD

**Issue by kels271828** [amath574w2015/am574-group06#1]
The initial conditions in the README don't match what I actually used in the code for the PW method. I was trying to use the second set of conditions (p,v)_l = (0, 0) and (p,v)_r = (0.4, 0.3), but the zero values were giving me trouble with NaNs, so I switched to (p, pv)_l = (0.01, 0.01) and (p, pv) = (0.4, 0.12).  The problem is, the left state velocity ends up being 1 instead of close to zero, because I got confused with velocity and momentum.  So it really should be something like (p, pv)_1 = (0.01, 0).  But I think from looking at the paper that we shouldn't be starting with zero density anywhere anyway?

**Comment by rjleveque** [comment]
Running your code as currently set up in af6280c9, it seems like you get a region with negative velocity forming, perhaps due to $u_r$ not being large enough for the density on the right?  Brisa and I were just talking about this and it's not clear from the Aw-Rascle paper whether there's an "equilibrium velocity" for a given density in the sense used in the PW model, but if so, taking initial data that respects this might be best.

You might want to try a "traffic jam" problem where $\rho_r = \rho_max$ and $u_r = 0$ and the left state has lower density with cars approach the jam.  I'd expect a solution with a single shock wave in this case.

I notice you're using a Roe-type solver in `pw.f90`, is it clear that the $\hat A$ matrix you're using to define the eigenvectors/values satisfies the Roe condition?

**Comment by kels271828** [comment]
The code I have up now is only for the PW model, which does predict negative velocities. This is a problem with the model, and not with the code (which is the point of the Daganzo paper).  I haven't really started working on the AR model yet.  

For the Roe-solver, I was trying to modify what was in the book for the shallow water equations, I think, which had a very similar Jacobian.  But I think I might be able to get away with just evaluating the Jacobian at averaged values rather than Roe values.  I'm still a little fuzzy about the Roe solver stuff and was planning on going over that again tonight.

**Comment by rjleveque** [comment]
Are you sure this is the PW model?  I thought that model has a source term and an equilibrium velocity $V(\rho)$ rather than a pressure $p(\rho) = \rho^\gamma$.

**Comment by kels271828** [comment]
Yes, the PW model has source terms which include a relaxation term with the equilibrium velocity and then a diffusion term.  However, we think that the authors of the paper were considering the model in the case that both those terms vanish.  They also mentioned that they were using the pressure function for their model, but didn't say what they were using for the PW model.  We just assumed that they were using it for both.  (The PW model has both the pressure function, or "anticipation factor", and the equilibrium velocity).
